### ISSUE TITLE
lowpkg.info fails on CentOS 6.7 minion

### ISSUE THREAD

**Issue by redmcg** [saltstack/salt#28075]
When running lowpkg.info against a CentOS 6.7 minion I get the following error:
```
$ salt '*' lowpkg.info salt-minion
    The minion function caused an exception: Traceback (most recent call last):
      File "/usr/lib/python2.6/site-packages/salt/minion.py", line 1036, in _thread_return
        return_data = func(*args, **kwargs)
      File "/usr/lib/python2.6/site-packages/salt/modules/rpm.py", line 470, in info
        value = _pkg_time_to_iso(value)
      File "/usr/lib/python2.6/site-packages/salt/modules/rpm.py", line 409, in _pkg_time_to_iso
        ptime = time.strptime(pkg_time, '%a %d %b %Y %H:%M:%S %p %Z')
      File "/usr/lib64/python2.6/_strptime.py", line 454, in _strptime_time
        return _strptime(data_string, format)[0]
      File "/usr/lib64/python2.6/_strptime.py", line 328, in _strptime
        data_string[found.end():])
    ValueError: unconverted data remains:       Build Host: localhost
```
I discovered this is because the rpm command being executed produced the following ouput:
```
$ rpm -qi salt-minion
Name        : salt-minion                  Relocations: (not relocatable)
Version     : 2015.8.1                          Vendor: (none)
Release     : 1.el6                         Build Date: Wed 07 Oct 2015 13:08:52 PDT
Install Date: Sun 18 Oct 2015 22:06:15 PDT      Build Host: localhost
Group       : System Environment/Daemons    Source RPM: salt-2015.8.1-1.el6.src.rpm
Size        : 37112                            License: ASL 2.0
Signature   : RSA/SHA1, Wed 07 Oct 2015 14:28:23 PDT, Key ID 0e08a149de57bfbe
URL         : http://saltstack.org/
Summary     : Client component for Salt, a parallel remote execution system
Description :
The Salt minion is the agent component of Salt. It listens for instructions
from the master, runs jobs, and returns results back to the master.
```
Note that there is multiple 'keys' per line (for example: 'Name' and 'Relocations' appear on the first line) but the code is expecting one per line. As a result when it attempts to parse the date on the 'Install Date' line it is picking up the 'Build Host' key as well. This results in the `ValueError: unconverted data remains:       Build Host: localhost` error.

I downloaded the source for RPM and found that the '-i' option is a macro for the '--qf' option with a predefined format. There is also a comment stating that this is not necessarily consistent. It would therefore appear that relying on the '-i' option could result in undefined behaviour (where the resultant '--qf' is not as expected) and that a safer approach would be to use a hard-coded value for '--qf'.

**Comment by redmcg** [comment]
The change proposed within Pull Request #28076  results in the rpm output of:
```
$ rpm -q salt-minion --queryformat 'Name: %-27{NAME}\nRelocations: %|PREFIXES?{[%{PREFIXES} ]}:{(not relocatable)}|\nVersion: %-27{VERSION}\nVendor: %{VENDOR}\nRelease: %-27{RELEASE}\nBuild Date: %{BUILDTIME:date}\nInstall Date: %|INSTALLTIME?{%-27{INSTALLTIME:date}}:{(not installed)         }|\nBuild Host: %{BUILDHOST}\nGroup: %-27{GROUP}\nSource RPM: %{SOURCERPM}\nSize: %-27{LONGSIZE}\n%|LICENSE?{License: %{LICENSE}\n}|Signature: %|DSAHEADER?{%{DSAHEADER:pgpsig}}:{%|RSAHEADER?{%{RSAHEADER:pgpsig}}:{%|SIGGPG?{%{SIGGPG:pgpsig}}:{%|SIGPGP?{%{SIGPGP:pgpsig}}:{(none)}|}|}|}|\n%|PACKAGER?{Packager: %{PACKAGER}\n}|%|URL?{URL: %{URL}\n}|Summary: %{SUMMARY}\nDescription:\n%{DESCRIPTION}\n-----\n'
Name: salt-minion                
Relocations: (not relocatable)
Version: 2015.8.1                   
Vendor: (none)
Release: 1.el6                      
Build Date: Wed 07 Oct 2015 13:08:52 PDT
Install Date: Sun 18 Oct 2015 22:06:15 PDT
Build Host: localhost
Group: System Environment/Daemons 
Source RPM: salt-2015.8.1-1.el6.src.rpm
Size: 37112                      
License: ASL 2.0
Signature: RSA/SHA1, Wed 07 Oct 2015 14:28:23 PDT, Key ID 0e08a149de57bfbe
URL: http://saltstack.org/
Summary: Client component for Salt, a parallel remote execution system
Description:
The Salt minion is the agent component of Salt. It listens for instructions
from the master, runs jobs, and returns results back to the master.
-----
```
which is the same as the default queryformat but has each key and its value on a separate line. As a result `lowpkg.info` is now returning the expected result:
```
$ salt '*' lowpkg.info salt-minion
    ----------
    salt-minion:
        ----------
        build_date:
            2015-10-07T08:08:52Z
        build_host:
            localhost
        description:
            The Salt minion is the agent component of Salt. It listens for instructions
            from the master, runs jobs, and returns results back to the master.
        group:
            System Environment/Daemons
        install_date:
            2015-10-19T05:06:15Z
        license:
            ASL 2.0
        name:
            salt-minion
        release:
            1.el6
        relocations:
            (not relocatable)
        signature:
            RSA/SHA1, Wed 07 Oct 2015 09:28:23 PM UTC, Key ID 0e08a149de57bfbe
        size:
            37112
        source_rpm:
            salt-2015.8.1-1.el6.src.rpm
        summary:
            Client component for Salt, a parallel remote execution system
        url:
            http://saltstack.org/
        vendor:
            (none)
        version:
            2015.8.1
```

Note the the values used in the queryformat have been included within the rpm module since `2008-06-26 17:52:31 +0300` so I don't expect it to introduce any new incompatibilities.

**Comment by jfindlay** [comment]
@redmcg, thanks for the report.

**Comment by redmcg** [comment]
This should now be fixed with Pull Request #28076
